apps/frontend/src/pages/Charities.tsx

```diff
@@ -0,0 +1,258 @@
+import { useState, useEffect, ChangeEvent } from "react";
+import {
+  Typography,
+  Grid,
+  Switch,
+  FormControlLabel,
+  Box,
+  Avatar,
+  Paper,
+} from "@mui/material";
+import CharityCard from "../components/CharityCard";
+import SearchBar from "../components/searchBar";
+import { colors } from "../styles/colors";
+import _axios_instance from "../_axios_instance";
+import Spinner from "../components/Spinner";
+
+interface Organization {
+  name: string;
+  logoUrl: string;
+  donated: number;
+  receiving: boolean;
+}
+
+interface Charity {
+  id: string;
+  name: string;
+  description: string;
+  startDate: string;
+  endDate: string;
+  imageUrl: string;
+  organizations: Organization[];
+  funds: number;
+  listingsCount: number;
+}
+
+const Charities = () => {
+  const currentDate = new Date();
+  const [showPastEvents, setShowPastEvents] = useState(false);
+  const [timeRemaining, setTimeRemaining] = useState<string>("");
+  const [charities, setCharities] = useState<Charity[]>([]);
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const response = await _axios_instance.get("/charities");
+        setCharities(response.data);
+        setLoading(false);
+      } catch (error) {
+        console.error("Error fetching charities:", error);
+        setLoading(false);
+      }
+    };
+    fetchData();
+  }, []);
+
+  const currentCharity = Array.isArray(charities)
+    ? charities.find((charity) => new Date(charity.endDate) >= currentDate)
+    : undefined;
+
+  const pastCharities = Array.isArray(charities)
+    ? charities.filter((charity) => new Date(charity.endDate) < currentDate)
+    : [];
```

Comment:
null can be used here instead of a blank array, then you can just check if pastCharities is null or not. That would require an additional check on the pastCharities array though, so this implementation is probably fine too

Reply:
I'm going to leave it then